src/groupByKey.ts

```diff
@@ -2,6 +2,15 @@ type GroupsMap<T> = {
   [key: string]: T[];
 };
 
-export function groupByKey(items, key) {
-  // write code here;
+export function groupByKey<T>(items: T[], key: keyof T): GroupsMap<T> {
+  return items.reduce((result, item) => {
+    const keyValue = item[key] as unknown as string;
+
+    if (!result[keyValue]) {
+      result[keyValue] = [];
+    }
+    result[keyValue].push(item);
+
+    return result;
+  }, {} as GroupsMap<T>);
```

Comment:
The initial value for the `reduce` function is cast to `GroupsMap<T>`. However, this cast is unsafe because the object does not initially have any keys that map to arrays of `T`. It would be safer to start with an empty object and let TypeScript infer the type as it accumulates properties.